packages/edit-post/src/components/sidebar/post-link/index.js

```diff
@@ -104,8 +107,18 @@ function PostLink( {
 			<div className="edit-post-post-link__preview-link-container">
 				<ExternalLink
 					className="edit-post-post-link__link"
-					href={ postLink }
+					href={ isEditable ? permalink : postLink }
```

Comment:
If we do this, we might want to use `addQueryArg` to add the `preview` bool to `permalink` (in case the post is still in draft state).